src/test/regress/sql/with.sql

```diff
@@ -1083,3 +1083,41 @@ WITH cte AS (
 
 RESET optimizer;
 DROP TABLE d;
+
+-- Check if sharing is disabled for a SegmentGeneral CTE to avoid deadlock if CTE is
+-- executed with 1-gang and joined with Hashed
+SET optimizer = off;
+--start_ignore
+DROP TABLE IF EXISTS d;
+DROP TABLE IF EXISTS r;
+--end_ignore
+
+CREATE TABLE d (a int, b int) DISTRIBUTED BY (a);
+INSERT INTO d VALUES (1,2),(2,3);
+CREATE TABLE r (a int, b int) DISTRIBUTED REPLICATED;
+INSERT INTO r VALUES (1,2),(3,4);
+
+EXPLAIN (COSTS off)
+WITH cte AS (
+    SELECT count(*) a FROM r
+) SELECT * FROM cte JOIN (SELECT * FROM d JOIN cte USING (a) LIMIT 1) d_join_cte USING (a);
+
+WITH cte AS (
+    SELECT count(*) a FROM r
+) SELECT * FROM cte JOIN (SELECT * FROM d JOIN cte USING (a) LIMIT 1) d_join_cte USING (a);
+
+-- Check if sharing is still enabled for other cases
+ALTER TABLE r SET DISTRIBUTED BY (a);
+
+EXPLAIN (COSTS off)
+WITH cte AS (
+    SELECT count(*) a FROM r
+) SELECT * FROM cte JOIN (SELECT * FROM d JOIN cte USING (a) LIMIT 1) d_join_cte USING (a);
+
+WITH cte AS (
+    SELECT count(*) a FROM r
+) SELECT * FROM cte JOIN (SELECT * FROM d JOIN cte USING (a) LIMIT 1) d_join_cte USING (a);
```

Comment:
are we testing join hash distributed with hash distributed here? it seems to me that the previous test testing the same thing. do we really need this?

Reply:
> it seems to me that the previous test testing the same thing

No, he changed the distribution of table r in line 1110.

Reply:
I see. I mean this test
https://github.com/arenadata/gpdb/blob/17c7ac54d14f2bc3d2878165d563b84b175e460e/src/test/regress/sql/with.sql#L1072-L1082

Reply:
Oh yes, then the second test looks redundant.

Reply:
Replacing this test with the test of joining two `r`'s.